lib/graphql/subscriptions/anycable_subscriptions.rb

```diff
@@ -70,40 +73,88 @@ def initialize(serializer: Serialize, **rest)
       # An event was triggered.
       # Re-evaluate all subscribed queries and push the data over ActionCable.
       def execute_all(event, object)
+        execute_legacy(event, object) if config.handle_legacy_subscriptions
+
+        fingerprints = redis.zrange(FINGERPRINTS_PREFIX + event.topic, 0, -1)
+        return if fingerprints.empty?
+
+        fingerprint_subscription_ids = Hash[fingerprints.zip(
+          redis.pipelined do
+            fingerprints.map do |fingerprint|
+              redis.smembers(SUBSCRIPTIONS_PREFIX + fingerprint)
+            end
+          end
+        )]
+
+        fingerprint_subscription_ids.each do |fingerprint, subscription_ids|
+          execute_grouped(fingerprint, subscription_ids, event, object)
+        end
+
+        # Call to +trigger+ returns this. Convenient for playing in console
+        Hash[fingerprint_subscription_ids.map { |k,v| [k, v.size] }]
+      end
+
+      def execute_grouped(fingerprint, subscription_ids, event, object)
+        return if subscription_ids.empty?
+
+        # The fingerprint has told us that this response should be shared by all subscribers,
+        # so just run it once, then deliver the result to every subscriber
+        result = execute_update(subscription_ids.first, event, object)
```

Comment:
Unless I'm mistaken, at this point `subscription_ids.first` might have been unsubscribed which will to an error in `read_subscription` (because we'll try to deserialize an empty context).
Does that make sense?

Reply:
Here is the stacktrace:

```
no implicit conversion of nil into String

lib/json/common.rb:263 initialize |  
lib/json/common.rb:263 new |  
lib/json/common.rb:263 parse |  
lib/graphql/subscriptions/serialize.rb:20 load |  
graphql-anycable (50c4abd1ad74) lib/graphql/subscriptions/anycable_subscriptions.rb:171 block in read_subscription |  
graphql-anycable (50c4abd1ad74) lib/graphql/subscriptions/anycable_subscriptions.rb:170 tap |  
graphql-anycable (50c4abd1ad74) lib/graphql/subscriptions/anycable_subscriptions.rb:170 read_subscription |  
lib/graphql/subscriptions.rb:101 execute_update |  
graphql-anycable (50c4abd1ad74) lib/graphql/subscriptions/anycable_subscriptions.rb:102 execute_grouped |  
graphql-anycable (50c4abd1ad74) lib/graphql/subscriptions/anycable_subscriptions.rb:90 block in execute_all |  
graphql-anycable (50c4abd1ad74) lib/graphql/subscriptions/anycable_subscriptions.rb:89 each |  
graphql-anycable (50c4abd1ad74) lib/graphql/subscriptions/anycable_subscriptions.rb:89 execute_all |  
lib/graphql/subscriptions.rb:87 trigger
```

Reply:
Oh yes. And to make things worse It also might have been expired.